File: Javascript/Algorithms/1221.js
/**
 * @param {string} s
 * @return {number}
 */
var balancedStringSplit = function(s) {
  let count = 0, cur_char = s[0], track = 1;
  let i = 1, end = s.length;
  while (i < end) {
    let next_char = s[i];
    if (next_char != cur_char) {
      track--;
      if (track == 0) {
        count++;
        i++;
        if (i < end) {
          cur_char = s[i];
          track = 1;
        } else
          break;
      }
    } else
      track++;
    i++;
  }
  return count;
};

File: C/Algorithms/1221.c
int balancedStringSplit(char *s) {
  int count = 0, track = 1;
  char cur_char = *s++;

  while (*s) {
    char next_char = *s++;
    if (cur_char != next_char) {
      track--;
      if (track == 0) {
        count++;
        if (*s)
          cur_char = *s++;
        else
          break;
        track = 1;
      }
    } else
      track += 1;
  }
  return count;
}